ray-operator/controllers/ray/common/pod.go

```diff
@@ -411,9 +411,13 @@ func concatenateContainerCommand(nodeType rayiov1alpha1.RayNodeType, rayStartPar
 	}
 
 	if _, ok := rayStartParams["num-gpus"]; !ok {
-		gpu := resource.Limits["gpu"]
-		if !gpu.IsZero() {
-			rayStartParams["num-gpus"] = strconv.FormatInt(gpu.Value(), 10)
+		// Scan for resource keys ending with "/gpu" like "nvidia.com/gpu".
+		for resourceKey, resource := range resource.Limits {
+			if strings.HasSuffix(string(resourceKey), "/gpu") && !resource.IsZero() {
```

Comment:
ideally, we would want "/gpu", "memory", "num-cpu" etc. in a separate file as constants, and that way if they change, we wouldn't have to change them in every place where they are used.

This could be a different PR that does that.

Reply:
Good idea -- will add some constants in a follow-up.

Reply:
Is it always going to end with "/gpu"? Or is it ever possible for it to be just "gpu"?

Reply:
+1 to Sriram's comment -- what was the previous behavior here (checking resource.Limits["gpu"]), why was it implemented but not unit-tested and are we making breaking changes here?

Presumably if it was not tested we can change it, but why was it there in the first place?

Reply:
GPUs are always specified as extended resources. In practice, either "nvidia.com/gpu" or "amd.com/gpu".
It appears to have been an oversight. Technically, dropping the "/" from the suffix will keep backwards compatibility, so I'll do that.